### ISSUE TITLE
If class is aliased out of a module, explicitly listed members not rendered into documentation

### ISSUE THREAD

**Issue by femtomc** [mkdocstrings/python#100]
**Describe the bug**
Explicitly listing members, but their documentation is not being shown (and code which I'm executing using `markdown-exec` is not being shown).

**Expected behavior**
Expect the members to show up in the documentation.

**Screenshots**
![image](https://github.com/mkdocstrings/python/assets/34410036/09b19808-21b4-4194-80bc-172a8c680ccc)

**System (please complete the following information):**
```
"mkdocstrings-python @ git+https://github.com/pawamoy-insiders/mkdocstrings-python.git@1.5.0.1.2.0"
```
- Python: 3.11.3
- OS: Reproduced this for my documentation on Ubuntu, as well as MacOS.

**Additional context**
Here's my `mkdocstrings` settings from my `mkdocs.yml`:

```yaml
  - mkdocstrings:
      enabled: !ENV [ENABLE_MKDOCSTRINGS, true]
      custom_templates: templates
      default_handler: python
      handlers:
        python:
          options:
            show_source: true
            show_root_heading: true
            show_root_full_path: false
            show_symbol_type_heading: true
            show_symbol_type_toc: true
```

Here's a log when I serve the documentation:
```
nox > python -m pip install --constraint=.nox/docs-serve/tmp/requirements.txt mkdocs
nox > python -m pip install --constraint=.nox/docs-serve/tmp/requirements.txt 'mkdocs-material @ git+https://github.com/probcomp/mkdocs-material-insiders'
nox > python -m pip install --constraint=.nox/docs-serve/tmp/requirements.txt 'mkdocstrings-python @ git+https://github.com/pawamoy-insiders/mkdocstrings-python.git@1.5.0.1.2.0'
nox > mkdocs serve
INFO    -  DeprecationWarning: warning_filter doesn't do anything since MkDocs 1.2 and will be
           removed soon. All messages on the `mkdocs` logger get counted automatically.
             File
           "/home/femtomc/Research/genjax/.nox/docs-serve/lib/python3.11/site-packages/mkdocstrings/loggers.py",
           line 9, in <module>
               from mkdocs.utils import warning_filter
             File
           "/home/femtomc/Research/genjax/.nox/docs-serve/lib/python3.11/site-packages/mkdocs/utils/__init__.py",
           line 453, in __getattr__
               warnings.warn(
INFO    -  Building documentation...
INFO    -  Cleaning site directory
INFO    -  The following pages exist in the docs directory, but are not included in the "nav"
           configuration:
             - index.md
             - homepage.md
             - genjax/diff_jl.md
             - genjax/language_aperitifs.md
             - genjax/concepts/generative_functions.md
             - genjax/library/diff_prog/index.md
             - genjax/library/diff_prog/adev.md
             - genjax/library/diff_prog/state.md
             - genjax/library/generative_functions/distributions/coryx.md
             - genjax/library/generative_functions/distributions/gensp.md
             - genjax/library/inference/index.md
             - genjax/library/inference/is.md
INFO    -  Doc file 'index.md' contains an absolute link '/genjax/notebooks/index.html', it was
           left as is.
INFO    -  Doc file 'genjax/notebooks.md' contains an absolute link
           '/genjax/notebooks/index.html', it was left as is.
INFO    -  DeprecationWarning: invalid escape sequence '\s'
             File
           "/home/femtomc/Research/genjax/.nox/docs-serve/lib/python3.11/site-packages/griffe/agents/visitor.py",
           line 177, in get_module
               top_node = compile(self.code, mode="exec", filename=str(self.filepath),
           flags=ast.PyCF_ONLY_AST, optimize=1)
             File "/home/femtomc/Research/genjax/src/genjax/_src/core/datatypes/generative.py",
           line 1110, in
               """> Given a `key: PRNGKey` and arguments `x: Tuple`, the generative
INFO    -  DeprecationWarning: invalid escape sequence '\l'
             File
           "/home/femtomc/Research/genjax/.nox/docs-serve/lib/python3.11/site-packages/griffe/agents/visitor.py",
           line 177, in get_module
               top_node = compile(self.code, mode="exec", filename=str(self.filepath),
           flags=ast.PyCF_ONLY_AST, optimize=1)
             File "/home/femtomc/Research/genjax/src/genjax/_src/core/datatypes/generative.py",
           line 1227, in
               """> Given a `key: PRNGKey`, a choice map indicating constraints ($u$),
INFO    -  DeprecationWarning: invalid escape sequence '\l'
             File
           "/home/femtomc/Research/genjax/.nox/docs-serve/lib/python3.11/site-packages/griffe/agents/visitor.py",
           line 177, in get_module
               top_node = compile(self.code, mode="exec", filename=str(self.filepath),
           flags=ast.PyCF_ONLY_AST, optimize=1)
             File "/home/femtomc/Research/genjax/src/genjax/_src/core/datatypes/generative.py",
           line 1266, in
               """> Given a `key: PRNGKey`, a complete choice map indicating
WARNING:jax._src.xla_bridge:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
INFO    -  Documentation built in 3.77 seconds
```

**Comment by femtomc** [comment]
The above strikes me as weird...

Seemingly, `griffe` gets to the source (here's another example):
![image](https://github.com/mkdocstrings/python/assets/34410036/2b769336-92e4-4b8e-acba-18d76cf66c26)

So e.g. it resolves the alias to get the source code -- but `mkdocstrings` is having issues. If I explicitly change the documentation to refer to the "inner" module (e.g. in my project, code which I'm hiding under a `_src` directory) -- it works again.

**Comment by pawamoy** [comment]
Hi @femtomc, thanks for the report! I'll investigate with what you gave me and come back to you if I need more info :slightly_smiling_face:

**Comment by pawamoy** [comment]
You're right, that highly possible. Just after our email exchange, I released mkdocstrings-python Insiders 1.5.1.1.3.0, you could try with that version, I think it will fix your issue.

**Comment by femtomc** [comment]
Okay will try! One moment, thanks!

**Comment by femtomc** [comment]
@pawamoy Should I set my inner (original) dataclasses `.public` attribute to `True`?

**Comment by pawamoy** [comment]
> @pawamoy Should I set my inner (original) dataclasses .public attribute to True?

If you refer to what is described in [`Attribute.is_public`](https://mkdocstrings.github.io/griffe/reference/griffe/?h=public#griffe.Attribute.is_public), quoting:

> If the object's public attribute is set (boolean), return its value.

...then no, by "object" here we refer to the Griffe object (Module/Class/Function/Attribute), not the actual object that is represented.

The `public` attribute (which by the way is undocumented I realize, will fix) is only meant to "force" an object to be considered public, and is only meant to be used by Griffe extensions, for example to support decorators that mark an object as public.

> be public if I export them via __all__

Yes! If you declare `__all__`, it is used to know if something is public. When it's not declared, Griffe falls back on additional logic.

**Comment by femtomc** [comment]
I think either (a) something might still be wrong or (b) I'm misunderstanding how to document:

![image](https://github.com/mkdocstrings/python/assets/34410036/030d7480-e72d-43fa-bca5-e3f787e81952)

Here, in `GenerativeFunction` -- I actually define the members and documentation which are showing up in `JAXGenerativeFunction`

In the markdown source:

```
::: genjax.core.GenerativeFunction
    options:
      members: 
        - simulate
        - propose
        - importance
        - assess
        - update

The interface definitions of generative functions often interact with JAX tracing machinery. GenJAX does not strictly impose this requirement, but does provide a generative function subclass called `JAXGenerativeFunction` which provides default compatibility definitions for JAX tracing.

Other generative function languages which utilize callee generative functions can enforce JAX compatibility by typechecking on `JAXGenerativeFunction`.

::: genjax.core.JAXGenerativeFunction
```

**Comment by pawamoy** [comment]
Thanks for the update. Is your repo public? Otherwise I'll run some tests with dummy code :+1:

**Comment by femtomc** [comment]
@pawamoy It is not public, but I could invite you if you want to do tests with it locally.

**Comment by femtomc** [comment]
That's fine! If you want to interact with me about the repo, let's do so via an issue which you start at the repo itself :) thanks!

**Comment by pawamoy** [comment]
This was a regression in the `children.html` template. The conditions that checked whether members were specified or the object was public were wrongly written. This is fixed in mkdocstrings-python 1.5.2 and Insiders 1.5.2.1.3.0.
